numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -3128,5 +3129,9 @@ init_stringdtype_ufuncs(PyObject *umath)
         return -1;
     }
 
+    if (init_stringdtype_sorts() < 0) {
+        return -1;
+    }
```

Comment:
maybe you can avoid the numpy initialization problem you're having if you move this into `init_string_dtype` over in the `dtype.c` file in the `multiarray/stringdtype` folder.